### PR TITLE
Add a reference for the LRScheduler class

### LINKED ISSUE #127884: lr_scheduler()
### 📚 The doc issue

https://pytorch.org/docs/stable/optim.html#module-torch.optim.lr_scheduler
When clicking on this link it takes you to a blank, bottom of the page section of torch.optim. The link for this function inside torch.optim just under the section "How to adjust learning rate", also links to this empty section...

Here is the plain link to the page w/o the stuff after .html
https://pytorch.org/docs/stable/optim.html

### Suggest a potential alternative/fix

Please make a section regarding how this function works on this page, and what parameters can go into the function optim.lr_scheduler(). 

cc @svekars @brycebortree @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

### PR DESCRIPTION
**Pull request by spzala**
The `LRScheduler` class provides methods to adjusts the learning rate during optimization (as updated in this PR). Also, as a note, all the classes of lr_scheduluer are already provided in the `How to adjust learning rate` section.

Fixes #127884

### CODE REVIEW

**Comment by janeyx99** [comment]
@spzala do you happen to have a before + after screenshot for the docs?

**Comment by janeyx99** [comment]
Looks like the docs test is failing:

```
2024-08-12T18:17:04.6538184Z /opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:docstring of torch.optim.lr_scheduler:1: WARNING: duplicate object description of torch.optim.lr_scheduler, other instance in optim, use :noindex: for one of them
2024-08-12T18:17:04.6540631Z looking for now-outdated files... none found
2024-08-12T18:17:04.8679277Z pickling environment... done
2024-08-12T18:17:04.8687738Z checking consistency... /var/lib/jenkins/workspace/docs/source/optim.lr_scheduler.rst: WARNING: document isn't included in any toctree
2024-08-12T18:17:04.8688809Z done
```

**Comment by spzala** [comment]
> @spzala do you happen to have a before + after screenshot for the docs?

Hi @janeyx99 yes, please find them attached. The currently behavior (attached - current-behavior.png) is (as described in the issue) this link https://pytorch.org/docs/stable/optim.html#module-torch.optim.lr_scheduler takes you to a blank, bottom of the page section of torch.optim. The screenshot from this fix (attached - after-this-pr.png) was captured from `file:///Users/..../docs/build/html/optim.lr_scheduler.html#module-torch.optim.lr_scheduler ` Thanks!!
after-this-pr
current-behavior

**Comment by spzala** [comment]
> Looks like the docs test is failing:
> 
> ```
> 2024-08-12T18:17:04.6538184Z /opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:docstring of torch.optim.lr_scheduler:1: WARNING: duplicate object description of torch.optim.lr_scheduler, other instance in optim, use :noindex: for one of them
> 2024-08-12T18:17:04.6540631Z looking for now-outdated files... none found
> 2024-08-12T18:17:04.8679277Z pickling environment... done
> 2024-08-12T18:17:04.8687738Z checking consistency... /var/lib/jenkins/workspace/docs/source/optim.lr_scheduler.rst: WARNING: document isn't included in any toctree
> 2024-08-12T18:17:04.8688809Z done
> ```

Oh, thanks for sharing! Let me work on it. I didn't think the test will fail :( Also, I am no sure why CLA is failing.

**Comment by spzala** [comment]
@janeyx99 I am still working on the doc test warnings but just updated commit to test CLA which seems good now. Thanks!

**Comment by spzala** [comment]
Hello @janeyx99 - the PR is ready for review now. I initially overthought but I think all we need to fix the issue is what I have changed in the PR update :) Thanks!

**Comment by janeyx99** [comment]
Nice! Running the CI to generate the docs so I could take a look. Thanks for working on this!

**Review by janeyx99 — APPROVED** [review]
thank u for figuring this out! by the way, i think swa_utils has the same issue ;) and i’d review a PR fixing that too

**Comment by spzala** [comment]
> thank u for figuring this out! by the way, i think swa_utils has the same issue ;) and i’d review a PR fixing that too

Thanks @janeyx99 Ah, haha, yes, I see that issue. Sure, I will work on creating a PR for that :) No PR's already open for it right? Thanks again!

**Comment by spzala** [comment]
> thank u for figuring this out! by the way, i think swa_utils has the same issue ;) and i’d review a PR fixing that too

@janeyx99 - for the swa_utils related fix, I have opened https://github.com/pytorch/pytorch/pull/133393 Thanks!
